x-pack/plugin/src/test/resources/rest-api-spec/api/data_frame.put_data_frame_transform.json

```diff
@@ -0,0 +1,21 @@
+{
+  "data_frame.put_data_frame_transform": {
+    "documentation": "TODO",
+    "methods": [ "PUT" ],
+    "url": {
+      "path": "/_data_frame/transforms/{transform_id}",
+      "paths": [ "/_data_frame/transforms/{transform_id}", "/_data_frame/transforms" ],
+      "parts": {
+        "transform_id": {
+          "type": "string",
+          "required": false,
+          "description": "The id of the new transform. If not provided, must be included in the body"
```

Comment:
> If not provided, must be included in the body

Should be:

The id may also optionally be provided in the body, but must match the id in the URL

(Or alternatively we could just omit this as a minor detail - what does put_job say about this?  The reason for this functionality is to make it easier to take config returned from the GET endpoint and send it back unaltered to the PUT endpoint.)